hedgehog/src/Hedgehog/Internal/Tree.hs

```diff
@@ -107,21 +107,23 @@ instance Functor m => Functor (Tree m) where
   fmap f =
     Tree . fmap (fmap f) . runTree
 
-instance Monad m => Applicative (Node m) where
-  pure =
-    return
-  (<*>) =
-    ap
+instance Applicative m => Applicative (Node m) where
+  pure x =
+    Node x []
+  (<*>) (Node ab tabs) na@(Node a tas) =
+    Node (ab a) $
+      map (fmap ab) tas ++ map (<*> (fromNode na)) tabs
 
-instance Monad m => Applicative (Tree m) where
+instance Applicative m => Applicative (Tree m) where
   pure =
-    return
-  (<*>) =
-    ap
+    Tree . pure . pure
+  (<*>) (Tree mab) (Tree ma) =
+    Tree $
+      (\nab na -> nab <*> na) <$> mab <*> ma
```

Comment:
I don't like the eta expansion of `(<*>)`
I would change this to `liftA2 (<*>) mab ma`

Reply:
Done! :)